app.rb

```diff
@@ -1,9 +1,32 @@
 require 'sinatra/base'
+require 'sinatra/reloader'
+require './lib/peeps'
 
 class Chitter < Sinatra::Base
+  configure :development do
+    register Sinatra::Reloader
+  end
+
   get '/test' do
     'Test page'
   end
 
+  get '/' do
+    ENV
+
+    @peeps = Peeps.all
+    erb(:index)
+  end
+
+  get '/comment' do
+    erb(:comment)
+  end
+
+  post '/' do
+    Peeps.create(message = params[:message])
+
+    redirect '/'
+  end
+
```

Comment:
You're almost there with the routing but it's not quite RESTful.

REST is a convention used for creating routes in a database backed web application, such as this. Each of your routes should contain a resource (pluralised) like `peeps` then, when combined with the HTTP verbs you can create unique, very predictable, routes (you also need to add `new` and `edit` in some cases).

For example...

```ruby
get '/peeps' do 
# for retrieving all the peeps
end

get '/peeps/new' do
# to get the form for creating a new peep
end

post '/peeps' do
# used when the new peep form is submitted
end

get '/peeps/edit' do
# used to get the form for editing a peep, if that is possible
end

put '/peeps/:id' do
# for when the edit peep form is submitted
# note that we also need the peep id here since we're going to update a specific peep
end
```

Reply:
Using the above pattern, you can predict what the routes would be for any new resource, like `Post` or `Space` or `Event`... and so on

Reply:
Deviating from this convention can lead to confusion.. for example when I saw this...

```ruby
get '/comment' do
    erb(:comment)
  end
```

I thought that it was possible to for users to add comments to a peep.